src/packet/packet_data.rs

```diff
@@ -1,32 +1,26 @@
-use super::header::HeaderParser;
-use super::RawPacketData;
-
+use std::io::Write;
 use error::NetworkResult;
 
 /// Contains the raw data this packet exists of. Note that a packet can be divided into separate fragments
 #[derive(Debug, Default)]
 pub struct PacketData {
-    parts: Vec<RawPacketData>,
+    parts: Vec<Vec<u8>>,
```

Comment:
vec of a vec? could you explain this more and why you choose the inner as a vec?

Reply:
Because I removed `RawPacketData` this was just some unnecessarily code which could be worked away. First I did it since I thought I could have slices here. But I could not accomplish that.